### ISSUE TITLE
fatal error: 'Flutter/Flutter.h' file not found

### ISSUE THREAD

**Issue by lyntree** [flutter/flutter#86997]
Hi Flutter Team,

Can you please re-look at the 'Flutter H' issue?  My app perfectly worked fine at the beginning of the month, and I've not touched it then. This time I decided to try and add in a new feature which is Firebase Auth / Fire Base Core.

Let's just say, everything broke, not just the app but both the tools (Gradle, and Xcodes Building Tool / CocoaPods).

I've tried to find every solution but I've posted this to the Google Dev forums and no one has responded and I am now stuck on what to do to move forward from the issue on IOS.

The weird thing about it..  it doesn't just affect 'one thing', as soon as you try to comment off a dependency, then it tells me it's another dependency and the list goes on... with the same Flutter H missing file..

I've been trying for over 24 hours now, to get this IOS mac emulator to run, but literally, it will not work.

I've explained what I've done in the Google Groups, there is nothing else online that I can try to get this working.  Can you please look into this, please ignore the 'integration tests' that broke a while ago and normally passes build but once again 'highlights the Flutter H issue'

https://groups.google.com/g/flutter-dev/c/DMZffnhpPAw 

```
```

```
```

</details>

**Comment by lyntree** [comment]
[FullOutput.txt](https://github.com/flutter/flutter/files/6873033/FullOutput.txt)

Sorry, I'm not sure if the file did load up as I can't see it listed, but all the commands you asked me to run It's within the file.

**Comment by TahaTesser** [comment]
Hi @lyntree 
Yur apps fails here
```console
[   +6 ms] executing: /usr/bin/arch -arm64e xcrun simctl spawn
21EEF94B-7292-4410-A225-04051023DB56 log stream --style json --predicate
eventType = logEvent AND processImagePath ENDSWITH "Runner" AND (senderImagePath
ENDSWITH "/Flutter" OR senderImagePath ENDSWITH "/libswiftCore.dylib" OR
processImageUUID == senderImageUUID) AND NOT(eventMessage CONTAINS ": could not
find icon for representation -> com.apple.") AND NOT(eventMessage BEGINSWITH
"assertion failed: ") AND NOT(eventMessage CONTAINS " libxpc.dylib ")
```

Did you make any changes to Runner?

To better address the issue, Can you please provide a complete reproducible minimal code sample without third-party 
 \and complete `flutter run --verbose` logs running the minimal code sample?
Thank you

**Comment by lyntree** [comment]
The Only change I made was upping the targeting deployment from IOS 09 to IOS 10, and had to force quite a few of the packages up to support that otherwise they would not work but other than that didn't do anything else.

I also had a weird issue in Xcode  / runner that when I did make these changes to the packages, they wouldn't save and ended up having to put a line in the runner file 'target' all one version.

Thank you for looking into this for me, its quite an annoying bug x

**Comment by TahaTesser** [comment]
Hi @lyntree 
We need reproducible code sample

> To better address the issue, Can you please provide a complete reproducible minimal code sample without third-party 
>  \and complete `flutter run --verbose` logs running the minimal code sample?
> Thank you

**Comment by lyntree** [comment]
[flutter.txt](https://github.com/flutter/flutter/files/6884180/flutter.txt)

Tried zipping the whole folder but I don't think Github will accept it, is there another way I could send this over to you?

**Comment by lyntree** [comment]
Sorry did not mean to close this.

**Comment by lyntree** [comment]
Might I add, ignore the https://github.com/florent37/Flutter-AssetsAudioPlayer issue, I've raised this also directly with that team as that's a bug that I seem to have for Android specifically, or 'was' but now I am seeing this also when I do the 'flutter run v' command.

**Comment by TahaTesser** [comment]
Hi @lyntree
as mentioned in https://github.com/flutter/flutter/issues/86997#issuecomment-887330092, the logs you provided are not enough to address the issue. 
Please provide reproducible steps from scratch without using third-party packages and  a complete reproducible minimal code sample and logs from this minimal code sample, not from your production app 

Thank you

**Comment by lyntree** [comment]
I'm sorry can you clarify how that I could do this if possible?  I developed my apps in production and I don't have any staging from different environments.

**Comment by TahaTesser** [comment]
@lyntree 
`flutter create myapp`
`cd myapp`
`flutter run -v` 
 
To begin with, add your first-party package or code that reproduces this issue or changes you made that caused this issue

**Comment by lyntree** [comment]
Okay, I'll respond in a few hours and see what else I can find

**Comment by lyntree** [comment]
Right Teha, I've done a bit more testing,  and as much as I would like to give you what you need, to be honest, I don't think that will work and it will break code and cause nothing to run, to be honest. 

What I have just tried doing is disabling as many dependencies, and I can confirm I think I have narrowed the issue down to the 'Path Provider' Plugin.

As I mentioned, the only difference between the original version of the app edited a month ago and this version this month is that I added Firebase Auth & Firebase Core, and I've ruled out these with the dependencies disabled.

Running "flutter pub get" in ad...                               1,124ms
Launching lib/main.dart on iPhone 12 in debug mode...
Running pod install...                                             10.0s
Running Xcode build...                                                  
Xcode build done.                                            6.1s
Failed to build iOS app
Error output from Xcode build:
↳
    objc[91394]: Class AMSupportURLConnectionDelegate is implemented in both ?? (0x1f654c188) and ??
    (0x113ab82b8). One of the two will be used. Which one is undefined.
    objc[91394]: Class AMSupportURLSession is implemented in both ?? (0x1f654c1d8) and ?? (0x113ab8308). One of
    the two will be used. Which one is undefined.
    ** BUILD FAILED **


Xcode's output:
↳
    error: the following command failed with exit code 1 but produced no further output
    CompileC
    /Users/lyntree/Library/Developer/Xcode/DerivedData/Runner-akxbrkvjvnlgdadvzftqvwkmsshf/Build/Intermediates.no
    index/Pods.build/Debug-iphonesimulator/path_provider.build/Objects-normal/x86_64/FLTPathProviderPlugin.o
    **/Users/lyntree/flutter/.pub-cache/hosted/pub.dartlang.org/path_provider-2.0.2/ios/Classes/FLTPathProviderPlug
    in.m normal x86_64 objective-c com.apple.compilers.llvm.clang.1_0.compiler**
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Analyzing workspace
    note: Constructing build description
    note: Build preparation complete

Could not build the application for the simulator.
Error launching application on iPhone 12.
mac:ad lyntree$ flutter run
Launching lib/main.dart on iPhone 12 in debug mode...
Running pod install...                                             10.9s
Running Xcode build...                                                  
Xcode build done.                                            8.0s
Failed to build iOS app
Error output from Xcode build:
↳
    objc[92001]: Class AMSupportURLConnectionDelegate is implemented in both ?? (0x1f654c188) and ??
    (0x11bad02b8). One of the two will be used. Which one is undefined.
    objc[92001]: Class AMSupportURLSession is implemented in both ?? (0x1f654c1d8) and ?? (0x11bad0308). One of
    the two will be used. Which one is undefined.
    ** BUILD FAILED **


Xcode's output:
↳
    In file included from
    /Users/lyntree/flutter/.pub-cache/hosted/pub.dartlang.org/path_provider-2.0.2/ios/Classes/FLTPathProviderPlug
    in.m:5:
    /Users/lyntree/flutter/.pub-cache/hosted/pub.dartlang.org/path_provider-2.0.2/ios/Classes/FLTPathProviderPlug
    in.h:5:9: fatal error: 'Flutter/Flutter.h' file not found
    #import <Flutter/Flutter.h>
            ^~~~~~~~~~~~~~~~~~~
    1 error generated.
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Analyzing workspace
    note: Constructing build description
    note: Build preparation complete

Could not build the application for the simulator.
Error launching application on iPhone 12.


So just in case..  I completely removed the folders in the pub cache all of them for path provider, and now the issue has jumped over to the integration testing

''''

 1 error generated.
    In file included from
    /Users/lyntree/flutter/packages/integration_test/ios/Classes/IntegrationTestIosTest.
    m:6:
    /Users/lyntree/flutter/packages/integration_test/ios/Classes/IntegrationTestPlugin.h
    :5:9: fatal error: 'Flutter/Flutter.h' file not found
    #import <Flutter/Flutter.h>
            ^~~~~~~~~~~~~~~~~~~
    1 error generated.
    In file included from
    /Users/lyntree/flutter/packages/integration_test/ios/Classes/IntegrationTestPlugin.m
    :5:
    /Users/lyntree/flutter/packages/integration_test/ios/Classes/IntegrationTestPlugin.h
    :5:9: fatal error: 'Flutter/Flutter.h' file not found
    #import <Flutter/Flutter.h>
            ^~~~~~~~~~~~~~~~~~~
    1 error generated.
    In file included from
    /Users/lyntree/flutter/packages/integration_test/ios/Classes/IntegrationTestIosTest.
    m:6:
    /Users/lyntree/flutter/packages/integration_test/ios/Classes/IntegrationTestPlugin.h
    :5:9: fatal error: 'Flutter/Flutter.h' file not found
    #import <Flutter/Flutter.h>

'''

What can you make from this? Do I remove the integration testing to see what happens to?

**Comment by TahaTesser** [comment]
Hi @lyntree 

>    /Users/lyntree/flutter/.pub-cache/hosted/pub.dartlang.org/path_provider-2.0.2/ios/Classes/FLTPathProviderPlug
>     in.h:5:9: fatal error: 'Flutter/Flutter.h' file not found

For this, try 
`flutter clean`
`flutter pub upgrade`
`flutter pub get`

Can you please provide make sure you're using integration_test from SDK 
```
  integration_test:
    sdk: flutter
```

Please try without Firebase, for issues related to Firebase, please file an issue on their dedicated GitHub.

**Comment by giorgio79** [comment]
I just spent two days with this issue. There are tons of posts
Here is the only solution that worked https://stackoverflow.com/a/68191223/3921848
Looks like Flutter upgrade to the latest V2 messes up the iOS folder...

**Comment by lyntree** [comment]
Thanks, TehaTesser and Giorgia79, I have tried both of these solutions and that's why I have come here.  I even came across this issue last night and tried it - https://github.com/flutter/flutter/issues/70895 - no better result.

So far I've ruled out:

- Issues with Dependencies & Firebase Auth & Firebase Core
- Changing Target Deployment Version from 8/9 - 10.0 (IOS)

I do agree with @giorgio79, whatever Flutter has released in the last build has broken both 'Build tools' Gradle & Cocoapods.

What I would recommend to the Flutter if the team if they could please include in their 'releases' information about full requirements for the releases for the next flutter versions that would be helpful:

- Gradle Version (Required)
- CocoaPods Version (Required)
- Java Version / Open JKD Version

I will keep digging today to see if I can find a result at least with either of the version.

**Comment by lyntree** [comment]
If anyone should be following this,  I have now raised it with CocoaPods as I think the issue is there, @TahaTesser  if you could please work with cocoa pods to find out this issue as I have now other Flutter developers contacting me via the groups, which some are saying 'its been weeks' they have been stuck with this issue

https://github.com/CocoaPods/CocoaPods/issues/10821

**Comment by TahaTesser** [comment]
Hi @lyntree 
Sorry that you're going through this but you still haven't provided logs without firebase and without third-party packages, just with path_provider.

Please try a new project with path_provider if that helps

**Comment by lyntree** [comment]
Hi Andy,

I tried doing this with a brand new app this morning and the path provider importing just this and it worked in the simulator which is fair enough but I don't see the point of providing a sample of my code which will break the application anyway. 

The weirdest thing about this is, that I'm not even using it as an import, it's being imported from another package which I have no idea which one I had to add it to.

I even tried 'Deleting the entire IOS' folder this morning and replacing it with a 'sample app' which slightly made the error go but then as soon as I started calling the Pods,  the error came back.

I shouldn't have to announce on my own business to call the other flutter developers who are stuck having this issue onto Discord to try and brainstorm literally between us all which is causing this issue. 

Once we have got to the issue, we will come back to Flutter and I hope for Flutters sake they will release a fix in the next release because its quite an annoying bug

https://discord.gg/GxYWyftr

**Comment by lyntree** [comment]
Right after a few tests this morning, I've resulted in this, included are the logs.

Latest Version of Flutter  & and downgraded to Flutter 2.2.3

-Test 1 - Sample 'Flutter Create 'ABC'  (Include Provider In Yaml) - No Error... 
-Test 2 - Sample 'Flutter Create 'ABC' (Include Provider in Yaml + CocoaPods) - Boom Flutter H Issue with Path Provider!

Could we now get someone to specifically look at the 'Path Provider' dealing with CocoaPods, and maybe this person can work with CocoaPods directly to get this issue as I suspect it could be a compiler issue possibly but I don't understand the errors?

Clone the sample folder and test it yourself - https://github.com/lyntree/FlutterHIssue

That's the logs it was giving me.

[FlutterPathProvider.txt](https://github.com/flutter/flutter/files/6911139/FlutterPathProvider.txt)

**Comment by TahaTesser** [comment]
Hi @lyntree 
I just tried your project, got the same error but after deleting Podfile and Pod folder, executed `flutter clean`, `flutter pub get`, and `flutter run` and app launches fine, also tried with ` platform :ios, '10.0'`, the app still runs fine 

Here is the update project
[FlutterHIssue_working.zip](https://github.com/flutter/flutter/files/6915439/FlutterHIssue_working.zip)

```
Launching lib/main.dart on Taha’s iPhone in debug mode...
lib/main.dart:1
Automatically signing iOS for device deployment using specified development team in Xcode project: X8NNQ9CYL2
Xcode build done.                                           13.6s
[VERBOSE-2:FlutterObservatoryPublisher.mm(101)] Failed to register observatory port with mDNS with error -65555.
[VERBOSE-2:FlutterObservatoryPublisher.mm(103)] On iOS 14+, local network broadcast in apps need to be declared in the app's Info.plist. Debug and profile Flutter apps and modules host VM services on the local network to support debugging features such as hot reload and DevTools. To make your Flutter app or module attachable and debuggable, add a '_dartobservatory._tcp' value to the 'NSBonjourServices' key in your Info.plist for the Debug/Profile configurations. For more information, see https://flutter.dev/docs/development/add-to-app/ios/project-setup#local-network-privacy-permissions
[VERBOSE-2:FlutterObservatoryPublisher.mm(101)] Failed to register observatory port with mDNS with error -65555.
[VERBOSE-2:FlutterObservatoryPublisher.mm(103)] On iOS 14+, local network broadcast in apps need to be declared in the app's Info.plist. Debug and profile Flutter apps and modules host VM services on the local network to support debugging features such as hot reload and DevTools. To make your Flutter app or module attachable and debuggable, add a '_dartobservatory._tcp' value to the 'NSBonjourServices' key in your Info.plist for the Debug/Profile configurations. For more information, see https://flutter.dev/docs/development/add-to-app/ios/project-setup#local-network-privacy-permissions
```

<details> 
<summary>flutter doctor -v</summary> 
	
```console
[✓] Flutter (Channel stable, 2.2.3, on macOS 12.0 21A5294g darwin-x64, locale en-GB)
    • Flutter version 2.2.3 at /Users/tahatesser/Code/flutter_stable
    • Framework revision f4abaa0735 (5 weeks ago), 2021-07-01 12:46:11 -0700
    • Engine revision 241c87ad80
    • Dart version 2.13.4

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/tahatesser/Code/SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Users/tahatesser/Code/SDK
    • Java binary at: /Users/tahatesser/Code/openjdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_292-b10)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode-beta.app/Contents/Developer
    • Xcode 13.0, Build version 13A5201i
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[✓] VS Code (version 1.58.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.24.0

[✓] Connected device (4 available)
    • Taha’s iPhone (mobile) • 00008020-001059882212002E            • ios            • iOS 15.0
    • iPhone 12 (mobile)     • 3C173470-A5A2-45F2-BF7B-79B8218E717C • ios            • com.apple.CoreSimulator.SimRuntime.iOS-15-0 (simulator)
    • macOS (desktop)        • macos                                • darwin-x64     • macOS 12.0 21A5294g darwin-x64
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome 92.0.4515.107

! Doctor found issues in 1 category.

```

</details>

Please let run this project

**Comment by Sheikh-X** [comment]
I'm having a similar issue, I did this already and it still didn't build for iOS, throws up the flutter.h file not found error.

**Comment by Sheikh-X** [comment]
> Hi @lyntree
> I just tried your project, got the same error but after deleting Podfile and Pod folder, executed `flutter clean`, `flutter pub get`, and `flutter run` and app launches fine, also tried with ` platform :ios, '10.0'`, the app still runs fine
> 
> Here is the update project
> [FlutterHIssue_working.zip](https://github.com/flutter/flutter/files/6915439/FlutterHIssue_working.zip)
> 
> ```
> Launching lib/main.dart on Taha’s iPhone in debug mode...
> lib/main.dart:1
> Automatically signing iOS for device deployment using specified development team in Xcode project: X8NNQ9CYL2
> Xcode build done.                                           13.6s
> [VERBOSE-2:FlutterObservatoryPublisher.mm(101)] Failed to register observatory port with mDNS with error -65555.
> [VERBOSE-2:FlutterObservatoryPublisher.mm(103)] On iOS 14+, local network broadcast in apps need to be declared in the app's Info.plist. Debug and profile Flutter apps and modules host VM services on the local network to support debugging features such as hot reload and DevTools. To make your Flutter app or module attachable and debuggable, add a '_dartobservatory._tcp' value to the 'NSBonjourServices' key in your Info.plist for the Debug/Profile configurations. For more information, see https://flutter.dev/docs/development/add-to-app/ios/project-setup#local-network-privacy-permissions
> [VERBOSE-2:FlutterObservatoryPublisher.mm(101)] Failed to register observatory port with mDNS with error -65555.
> [VERBOSE-2:FlutterObservatoryPublisher.mm(103)] On iOS 14+, local network broadcast in apps need to be declared in the app's Info.plist. Debug and profile Flutter apps and modules host VM services on the local network to support debugging features such as hot reload and DevTools. To make your Flutter app or module attachable and debuggable, add a '_dartobservatory._tcp' value to the 'NSBonjourServices' key in your Info.plist for the Debug/Profile configurations. For more information, see https://flutter.dev/docs/development/add-to-app/ios/project-setup#local-network-privacy-permissions
> ```
> 
> flutter doctor -v
> 
> Please run this project



> Hi @lyntree
> I just tried your project, got the same error but after deleting Podfile and Pod folder, executed `flutter clean`, `flutter pub get`, and `flutter run` and app launches fine, also tried with ` platform :ios, '10.0'`, the app still runs fine
> 
> Here is the update project
> [FlutterHIssue_working.zip](https://github.com/flutter/flutter/files/6915439/FlutterHIssue_working.zip)
> 
> ```
> Launching lib/main.dart on Taha’s iPhone in debug mode...
> lib/main.dart:1
> Automatically signing iOS for device deployment using specified development team in Xcode project: X8NNQ9CYL2
> Xcode build done.                                           13.6s
> [VERBOSE-2:FlutterObservatoryPublisher.mm(101)] Failed to register observatory port with mDNS with error -65555.
> [VERBOSE-2:FlutterObservatoryPublisher.mm(103)] On iOS 14+, local network broadcast in apps need to be declared in the app's Info.plist. Debug and profile Flutter apps and modules host VM services on the local network to support debugging features such as hot reload and DevTools. To make your Flutter app or module attachable and debuggable, add a '_dartobservatory._tcp' value to the 'NSBonjourServices' key in your Info.plist for the Debug/Profile configurations. For more information, see https://flutter.dev/docs/development/add-to-app/ios/project-setup#local-network-privacy-permissions
> [VERBOSE-2:FlutterObservatoryPublisher.mm(101)] Failed to register observatory port with mDNS with error -65555.
> [VERBOSE-2:FlutterObservatoryPublisher.mm(103)] On iOS 14+, local network broadcast in apps need to be declared in the app's Info.plist. Debug and profile Flutter apps and modules host VM services on the local network to support debugging features such as hot reload and DevTools. To make your Flutter app or module attachable and debuggable, add a '_dartobservatory._tcp' value to the 'NSBonjourServices' key in your Info.plist for the Debug/Profile configurations. For more information, see https://flutter.dev/docs/development/add-to-app/ios/project-setup#local-network-privacy-permissions
> ```
> 
> flutter doctor -v
> 
> Please run this project

I'm having a similar issue, I did this already and it still didn't build for iOS, throws up the flutter.h file not found error.

**Comment by lyntree** [comment]
Hi @Sheikh-X , We have to and the same issue, you're invited to join us on - https://discord.gg/GxYWyftr you'll find other developers on here with the same problem, and we're working between ourselves to rule out what we have done and what we haven't amongst different versions. 

Question @TahaTesser , with this solution, still not working for us, if we do find a fix, how likely could we get Flutter to release this ASAP?

**Comment by Sheikh-X** [comment]
> Hi @Sheikh-X , We have to and the same issue, you're invited to join us on - https://discord.gg/GxYWyftr you'll find other developers on here with the same problem, and we're working between ourselves to rule out what we have done and what we haven't amongst different versions.
> 
> Question @TahaTesser , with this solution, still not working for us, if we do find a fix, how likely could we get Flutter to release this ASAP?

Hi Lyntree, I joined the Discord already. Thanks.

**Comment by TahaTesser** [comment]
@lyntree 
Sample in https://github.com/flutter/flutter/issues/86997#issuecomment-890823065 doesn't run for you? It runs after I updating pod files. Please try update project

**Comment by lyntree** [comment]
Nope, we tried it doesn't that's the weird thing.

**Comment by lyntree** [comment]
Just to follow up on this, okay so we've narrowed it down to either a 'Symlink' issue or a 'Compile' issue,  is there any way we could get these things? 

The weird thing is that the files are there and the Symlinks, I have come across other 'Compile issues' with Path Provider but we would like to get down to the root cause a nip this issue in the but so to speak, is there a way we could possibly fix these to things?

**Comment by TahaTesser** [comment]
Hi @lyntree 
Your tried project with new Podfile generated from `flutter pub get` on two different MBP and both run fine on both

> Just to follow up on this, okay so we've narrowed it down to either a 'Symlink' issue or a 'Compile' issue

The issue sounds to be local rather than a bug with Flutter,  `fatal error: 'Flutter/Flutter.h' file not found ` doesn't like seem to be caused by one thing in particular. 

I will label for additional triage so we can find the root cause

**Comment by lyntree** [comment]
We have tried everything @TahaTesser , I really hate to keep pushing on this issue, but literally have scoured the internet to find a root cause,  the fact that the group started on discord with about 4/5 people, its quite shocking now to find almost 12/13 people on the discord group so i'm guessing its not just a small issue. 

If we could get this further looked into it would be great, apologies but I may have to also open up another issue with Flutter too for Android because I can't get that issue solved either.

**Comment by Hixie** [comment]
Can any of the other people who are running into this determine the cause?

Without us being able to reproduce the problem it's really hard for us to do anything about it, since we can't really know what the problem is.

One technique that can help is to slowly delete code bit by bit until the error goes away, deleting as much as possible so you end up with the smallest possible file that still reproduces the problem. Have you had any luck with a technique like that?

**Comment by lyntree** [comment]
I had another developer who has years of experience take a look at this yesterday via remote and he tried everything, but what he reckoned that it is, it could be a 'translucent issue, that possible something else could be causing it as a direct knock-on effect?

What we did notice is that 'No matter, if you install Path Provider, for a different version, the 'Complier' still seeks for this exact version for 'no reason 'path_provider: 2.0.2'.

For some reason, after we disabled this package in mine, flutter_neumorphic it somewhat worked but (it means that I have to disable the entire code of the radio player) which now... I have to rewrite it possible or go back to the plugin developer.

But I then even quiet this against the developer, because that made no sense because when I sent that 'Sample ABC' app via Github for you guys to clone, it failed and it literally only had 'Path Provider' and it failed to work???

There's only so many times we can all put a fix on the app until it probably completely breaks...

Flutter need to take a look at the Complier as I do feel the issue is there, but it's either related to cache, swift or symlinks. I have no idea and I can't really read Swift. 

Anyone is welcome to join us on Discord if it's easier for you to triage this problem.

I understand the Flutter Team are probably frustrated and judging from the image I got sent via another developer yesterday, seems like this issue is 'cropping' up now more of a frustration for Flutter.

<img width="1288" alt="Screenshot 2021-08-04 at 12 01 11" src="https://user-images.githubusercontent.com/71215061/128170166-c928d296-8018-451d-a326-3a2be605bb91.png">

**Comment by lyntree** [comment]
The only other thing I can think of is Flutter creating some kind of sharing device so that we can send you our repositories securely and you can run it on your side

**Comment by RomanTechnologies** [comment]
To add some context.

I helped @lyntree yesterday go through the issues that were being experienced, finding the cause to be 'transient errors' from a package that had a conflicting class type with the Flutter framework. In this particular case, the culprit was AnimatedScale which was defined in both the Flutter framework and also a neo-morphic package.

That issue did not rise to the surface until we cleared the Dart and Flutter pub cache's, removing any issue that was being faced from the latest release of path_provider. Although I suspect it was a mixture between symlinks, xcode cache, ide cache, flutter's cache and build folders that hadn't been cleared. Hence why others who tested the minimum reproducible repo could run it.

I don't feel this issue is Flutter framework specific, but more the typical code devil that can arise with multiple installation versions of Flutter, it's packages and reuse of previously compiled packages within the build folder.

So my suggestion is this for anyone else experiencing similar issues.

- flutter pub cache clean
- flutter clean (ensure build folder is removed)
- delete symlinks, pods and podfile.lock from the ios folder
- delete your pubspec.lock
- build again

If you are still having issues you can go further
- flutter pub cache repair
- delete Generated.xcconfig from ios/Flutter directory
- pod install (from the ios directory)
- build again

Good luck all.

**Comment by lyntree** [comment]
Thanks for that @RomanTechnologies , sorry I couldn't remember the name. I've not had the Flutter H Issue so far over the weekend so I will close this for the time being
